packages/web-components/src/progress-bar/progress-bar.spec.ts

```diff
@@ -0,0 +1,152 @@
+  });
+
+  test('should render an element with a `determinate` slot when a value is provided', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+            <fluent-progress-bar value="50"></fluent-progress-bar>
+        `;
+    });
+
+    const progress = element.locator('.progress');
+
+    await expect(progress).toHaveAttribute('slot', 'determinate');
+  });
+
+  test('should render an element with an `indeterminate` slot when no value is provided', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+            <fluent-progress-bar></fluent-progress-bar>
+        `;
+    });
+
+    const progress = element.locator('.progress');
+
+    await expect(progress).toHaveAttribute('slot', 'indeterminate');
+  });
+
+  test('should return the `percentComplete` property as a value between 0 and 100 when `min` and `max` are unset', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+            <fluent-progress-bar value="50"></fluent-progress-bar>
+        `;
+    });
+
+    await expect(element).toHaveJSProperty('percentComplete', 50);
+  });
+
+  test('should set the `percentComplete` property to match the current `value` in the range of `min` and `max`', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+            <fluent-progress-bar value="0"></fluent-progress-bar>
+        `;
+    });
+
+    await expect(element).toHaveJSProperty('percentComplete', 0);
+
+    await element.evaluate((node: ProgressBar) => {
+      node.value = 50;
+    });
+
+    await expect(element).toHaveJSProperty('percentComplete', 50);
+
+    await element.evaluate((node: ProgressBar) => {
+      node.value = 100;
+    });
+
+    await expect(element).toHaveJSProperty('percentComplete', 100);
+
+    await element.evaluate((node: ProgressBar) => {
+      node.max = 200;
+    });
+
+    await expect(element).toHaveJSProperty('percentComplete', 50);
+
+    await element.evaluate((node: ProgressBar) => {
+      node.min = 100;
+    });
+
+    await expect(element).toHaveJSProperty('percentComplete', 0);
+  });
+
+  // Fluent Specific propertiy tests
+  test('should set and retrieve the `thickness` property correctly', async () => {
+    await element.evaluate((node: ProgressBar) => {
+      node.thickness = 'large';
+    });
+
+    await expect(element).toHaveJSProperty('thickness', 'large');
+  });
+
+  test('should set and retrieve the `shape` property correctly', async () => {
+    await element.evaluate((node: ProgressBar) => {
+      node.shape = 'square';
+    });
+
+    await expect(element).toHaveJSProperty('shape', 'square');
+  });
+
+  test('should set and retrieve the `validationState` property correctly', async () => {
+    await element.evaluate((node: ProgressBar) => {
+      node.validationState = 'warning';
+    });
+
+    await expect(element).toHaveJSProperty('validationState', 'warning');
+  });
```

Comment:
I think we should test for each possible variation of these as they are all part of the public API.

Reply:
Fixed!